back/engines/commercial/id_bosa_fas/app/lib/id_bosa_fas/bosa_fas_omniauth.rb

```diff
@@ -31,10 +31,11 @@ def omniauth_setup(configuration, env)
       options[:response_type] = :code
       options[:state] = true
       options[:nonce] = true
-      options[:issuer] = "https://#{host}"
+      options[:issuer] = "https://#{host}/fas/oauth2"
       options[:acr_values] = 'urn:be:fedict:iam:fas:Level450'
       options[:send_scope_to_token_endpoint] = false
       options[:client_signing_alg] = :RS256
+      options[:client_jwk_signing_key] = File.read(::IdBosaFas::Engine.root.join('config', 'bosa_fas_jwks.json'))
```

Comment:
Were you able to test that this works? I think I attempted something similar, by providing the jwks_uri on line 49 that links to the same JSON file:

```
   ENVIRONMENTS = {
      'integration' => {
        host: 'idp.iamfas.int.belgium.be',
        jwks_uri: 'https://idp.iamfas.int.belgium.be/fas/oauth2/connect/jwk_uri'
      },
      'production' => {
        host: 'idp.iamfas.belgium.be',
        jwks_uri: 'https://idp.iamfas.belgium.be/fas/oauth2/connect/jwk_uri'
      }
    }
```

Also, this will most likely break for the integration environment, as they use a different JSON file for it.

Reply:
It might be that simply fixing the `issuer` might already fix the issue, but I cannot try that out myself (because the existing `jwks_uri` option links to the very same JSON file already, but I'm guessing here).

Reply:
> Also, this will most likely break for the integration environment, as they use a different JSON file for it.

Indeed, very good point! 

> Were you able to test that this works?

Yep, I managed to test it locally. It was a bit tricky to setup though and I had to hack a few things to make it work, so the test may not be 100% reliable.

Strangely enough I could not find any usage of the `jwks_uri` in `omniauth_openid_connect`. I came to the conclusion it's just not being used by the library.

> It might be that simply fixing the issuer might already fix the issue 

I'm pretty sure it's not enough because I actually did things the other way around (the order of the commits is reversed). I first added the public keys. Then the error message changed to complain that the access token was not  valid due to a mismatch between issuers.

Reply:
> Also, this will most likely break for the integration environment, as they use a different JSON file for it.

Gonna add the keys for the other environment.